### DISCUSSION TITLE
How to tell LLM to pass required parameters in a tool call?

### DISCUSSION THREAD

**Discussion by CodeNameR** [agno-agi/agno#1107] [Q&A]
We have a Toolkit where we have defined functions like get_available_slots which has required parameters like start_date, end_date etc. 

We are using OpenAI and it is making the tool call without passing the start_date or end_date or other parameters. 
Is there a way to define which parameters are required? We are currently getting errors like this one - 

Missing required argument [type=missing_argument, input_value=ArgsKwargs((), {'start_da... '2024-08-30T10:00:00'}), input_type=ArgsKwargs]\n    For further information visit https://errors.pydantic.dev/2.7/

When using OpenAI directly, we could specify required fields in json. How can we achieve the same with PhiData?
{
    "name": "get_delivery_date",
    "description": "Get the delivery date for a customer's order. Call this whenever you need to know the delivery date, for example when a customer asks 'Where is my package'",
    "parameters": {
        "type": "object",
        "properties": {
            "order_id": {
                "type": "string",
                "description": "The customer's order ID.",
            },
        },
        "required": ["order_id"],
        "additionalProperties": false,
    }
}

Thanks,
Ritesh

**Comment by CodeNameR** [comment]
@ashpreetbedi - Is this something you can help with?

**Comment by manthanguptaa** [comment]
Hey @CodeNameR, I am sharing an example code of a toolkit that you can take inspiration from to develop your tool. My first guess is you didn't register the function.
Code: https://github.com/phidatahq/phidata/blob/main/phi/tools/slack.py
